pkg/envoy/xds/set.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xds
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/pkg/envoy/api"
+
+	"github.com/golang/protobuf/proto"
+)
+
+// ResourceSet is a versioned set of resources.
+// A set associates a single version to all the resources it contains.
+// The version is monotonically increased for any change to the set.
+type ResourceSet interface {
+	// GetResources returns the current version of the resources with the given
+	// names.
+	// If lastVersion is not nil and the resources with the given names haven't
+	// changed since lastVersion, nil is returned.
+	// If resourceNames is empty, all resources are returned.
+	// Should not be blocking.
+	GetResources(ctx context.Context, typeURL string, lastVersion *uint64,
+		node *api.Node, resourceNames []string) (*VersionedResources, error)
+}
+
+// VersionedResources is a set of protobuf-encoded resources along with their
+// version.
+type VersionedResources struct {
```

Comment:
This seems out of place here, in between a bunch of related interfaces. I don't think we use it in this file so putting it at the end might be ok. If you have a reason for it here, though, ignore me.

Reply:
This is the return type of `GetResources` defined just above. Since that method is the only public interface that uses that type, I considered that `VersionedResources` was part of the definition of that interface.

Reply:
Indeed, you are correct.